cmd/root.go

```diff
@@ -54,6 +55,11 @@ func initConfig() {
 	flags.BindPFlags(rootCmd.Flags())
 	config.InitConfig(flags)
 	setupLoging(flags)
+	nc := wireguard.NewNCIface(config.Netclient(), config.GetNodes())
+	if err := nc.Create(); err != nil {
+		slog.Error("failed to create interface, is wireguard installed", "error", err)
```

Comment:
> is wireguard installed?

Reply:
this will cause duplicate interface creation calls on daemon startup...
ideally, a separate function to check if wireguard is installed would make more sense

Reply:
this will cause duplicate interface creation call on daemon startup...
ideally, a separate func to check if wireguard is installed would make more sense here.